src/test/java/com/elasticbox/jenkins/tests/PullRequestManagerTest.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * ElasticBox Confidential
+ * Copyright (c) 2019 All Right Reserved, ElasticBox Inc.
+ *
+ * NOTICE:  All information contained herein is, and remains the property
+ * of ElasticBox. The intellectual and technical concepts contained herein are
+ * proprietary and may be covered by U.S. and Foreign Patents, patents in process,
+ * and are protected by trade secret or copyright law. Dissemination of this
+ * information or reproduction of this material is strictly forbidden unless prior
+ * written permission is obtained from ElasticBox.
+ */
+
+package com.elasticbox.jenkins.tests;
+
+import com.elasticbox.jenkins.util.Condition;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import net.sf.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.kohsuke.github.GHHook;
+
+
+
+public class PullRequestManagerTest extends PullRequestTestBase {
+    private static final Logger LOGGER = Logger.getLogger(PullRequestManagerTest.class.getName() );
+
+    private long NEXTBUILD_TIMEOUT = 30;
+    private long NEXTBUILD_TIMEOUT_LONG = 60;
+    private int MIN_NECESSARY_EXECUTORS = 3;
+    private final String TEST_TRIGGER_PHRASE = "Jenkins test this please ";
+
+    @Before
+    @Override
+    public void setup() throws Exception {
+        setLoggerLevel("com.elasticbox.jenkins", Level.FINER);
+
+        int numExecutors = jenkinsRule.getInstance().getNumExecutors();
+        if(numExecutors < MIN_NECESSARY_EXECUTORS ){
+            LOGGER.warning("Consider increasing the number of executors from " + numExecutors
```

Comment:
This is a test, you specify the executors that you need. This logger is not required.